src/v/redpanda/admin/server.cc

```diff
@@ -1519,61 +1519,90 @@ void config_multi_property_validation(
         using config_properties_seq = std::vector<std::reference_wrapper<
           const config::property<std::optional<ss::sstring>>>>;
 
-        config_properties_seq properties{};
-
-        if (
-          updated_config.cloud_storage_credentials_source
-          == model::cloud_credentials_source::config_file) {
-            config_properties_seq s3_properties = {
-              std::ref(updated_config.cloud_storage_region),
-              std::ref(updated_config.cloud_storage_bucket),
-              std::ref(updated_config.cloud_storage_access_key),
-              std::ref(updated_config.cloud_storage_secret_key),
-            };
-
-            config_properties_seq abs_properties = {
-              std::ref(updated_config.cloud_storage_azure_storage_account),
-              std::ref(updated_config.cloud_storage_azure_container),
-              std::ref(updated_config.cloud_storage_azure_shared_key),
-            };
-
-            std::array<config_properties_seq, 2> valid_configurations = {
-              s3_properties, abs_properties};
-
-            bool is_valid_configuration = std::any_of(
-              valid_configurations.begin(),
-              valid_configurations.end(),
-              [](const auto& config) {
-                  return std::none_of(
-                    config.begin(), config.end(), [](const auto& prop) {
-                        return prop() == std::nullopt;
-                    });
-              });
+        switch (updated_config.cloud_storage_credentials_source.value()) {
+        case model::cloud_credentials_source::config_file:
+            [&] {
```

Comment:
i don't think you need the lambda here you can open a scope `case config_file: { ... }`